pathspec/pathspec.py

```diff
@@ -30,6 +30,14 @@ def __init__(self, patterns):
 		"""
 
 		self.patterns = patterns if isinstance(patterns, collections.Container) else list(patterns)
+		
+	def __eq__(self, other):
+		"""
+		Tests equality of this ``PathSpec`` with ``other`` based on the
+		regexs contained in their ``patterns``.
+		"""
+		paired_patterns = zip(self.patterns, other.patterns)
```

Comment:
You might need to use `zip_longest()` here to avoid equality when one is a "prefix" of the other.